zk-token-sdk/src/sigma_proofs/fee_proof.rs

```diff
@@ -367,24 +369,15 @@ impl FeeSigmaProof {
             return Err(ProofVerificationError::Deserialization.into());
         }
 
-        let bytes = array_ref![bytes, 0, 256];
-        let (Y_max_proof, z_max_proof, c_max_proof, Y_delta, Y_claimed, z_x, z_delta, z_claimed) =
-            array_refs![bytes, 32, 32, 32, 32, 32, 32, 32, 32];
+        let Y_max_proof = CompressedRistretto::from_slice(&bytes[..32]);
+        let z_max_proof = canonical_scalar_from_slice(&bytes[32..64])?;
+        let c_max_proof = canonical_scalar_from_slice(&bytes[64..96])?;
 
-        let Y_max_proof = CompressedRistretto::from_slice(Y_max_proof);
-        let z_max_proof = Scalar::from_canonical_bytes(*z_max_proof)
-            .ok_or(ProofVerificationError::Deserialization)?;
-        let c_max_proof = Scalar::from_canonical_bytes(*c_max_proof)
-            .ok_or(ProofVerificationError::Deserialization)?;
-
-        let Y_delta = CompressedRistretto::from_slice(Y_delta);
-        let Y_claimed = CompressedRistretto::from_slice(Y_claimed);
-        let z_x =
-            Scalar::from_canonical_bytes(*z_x).ok_or(ProofVerificationError::Deserialization)?;
-        let z_delta = Scalar::from_canonical_bytes(*z_delta)
-            .ok_or(ProofVerificationError::Deserialization)?;
-        let z_claimed = Scalar::from_canonical_bytes(*z_claimed)
-            .ok_or(ProofVerificationError::Deserialization)?;
+        let Y_delta = CompressedRistretto::from_slice(&bytes[96..128]);
+        let Y_claimed = CompressedRistretto::from_slice(&bytes[128..160]);
+        let z_x = canonical_scalar_from_slice(&bytes[160..192])?;
+        let z_delta = canonical_scalar_from_slice(&bytes[192..224])?;
+        let z_claimed = canonical_scalar_from_slice(&bytes[224..256])?;
```

Comment:
I'm going to approve this as-is, but I sort of wish we could use some constants like `SCALAR_LEN` or `RISTRETTO_POINT_LEN` to clean up some of these magic numbers. Let's mull on whether something like that would make things less brittle and/or more readable.

Reply:
Oh, awesome! I know we talked about that for some other numbers. Awesome if it will work here as well without arrayref